### PR TITLE
Add test enforcing `make requirements` was run

### LINKED ISSUE #21580: pip-compile produces different results on different operating systems
More context https://github.com/dimagi/commcare-hq/pull/21559#discussion_r210748977

It appears that ipython has different dependencies if you're on a mac vs linux. I think we should default to linux's requirements since that's what our servers run, but as of right now I don't know of a way to enforce that.

This isn't currently breaking anything, but it will probably create some package churn if we start using pip-sync, and could potentially break if a package requires conflicting dependencies on different platforms

fyi: @kaapstorm since you ran into this the other day

### PR DESCRIPTION
**Pull request by dannyroberts**
and that running it again on linux does not produce a git diff

Pulled out from https://github.com/dimagi/commcare-hq/pull/21612 (which contained this as an unrelated change, and turned out to be hairier than I thought) and then squashed into one commit.

Update: I decide to solve the issue from https://github.com/dimagi/commcare-hq/pull/21612 in this PR as well, by simply post-processing requirements.txt files to remove lines that match `^appnope==`, thus matching the file that would be generated on Linux. So now fixes https://github.com/dimagi/commcare-hq/issues/21580.

### CODE REVIEW

**Comment by dannyroberts** [comment]
This test should fail, and then if it does, I'll run `make requirements` and commit the change, and then the test should pass.

**Comment by dannyroberts** [comment]
This went pretty much as planned with some tweaks at the end. Personally pretty happy with this approach.

File: scripts/pip-post-compile.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+set -e
+
+bash scripts/_vendor/pip-post-compile.sh "$@"
```
Comment (emord) [inline scripts/pip-post-compile.sh]: Somewhat unrelated to this PR, but it looks like this file is generated by curling github and then saving it outside of our git repo. Is there a reason why we aren't including that file in our repository?

Comment (dannyroberts) [reply scripts/pip-post-compile.sh]: Yeah good question. You can follow my reasoning in this PR description: https://github.com/dimagi/commcare-hq/pull/21490#issue-207163955.

Comment (emord) [reply scripts/pip-post-compile.sh]: Your reasoning makes sense. I do have an aversion to `curl | bash` but this is about the same as installing packages from specific commits/versions

**Review by emord — APPROVED** [review]

File: scripts/pip-post-compile.sh (hunk 1 above)
Comment (dannyroberts) [reply scripts/pip-post-compile.sh]: Yeah, I know what you mean. The nice things I considered are that it's over https, so it's not going to be man-in-the-middled, and the url contains the commit hash, so it's not going to change out from under us. The only way (that my little brain can think of) for us to get and execute the wrong thing is for github's infrastructure to get compromised, which is the same failure mode as checking out our own code and running it in production.

**Pull request merged**